src/vcpu.rs

```diff
@@ -0,0 +1,219 @@
+
+    /// Reads the floating point state from the vCPU
+    ///
+    /// - Floating Point MMX Registers 0-7
+    /// - Floating Pointer Control, Status, Tag Registers (FCW, FSW, FTW)
+    /// - Floating point exception state (Last FIP, FOP, FCS, FDS, FDP)
+    /// - XMM Registers 0-15
+    /// - MXCSR Control and Status Register
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    fn get_fpu(&self) -> Result<FpuState>;
+
+    /// Writes the floating point state to the vCPU
+    ///
+    /// - Floating Point MMX Registers 0-7
+    /// - Floating Pointer Control, Status, Tag Registers (FCW, FSW, FTW)
+    /// - Floating point exception state (Last FIP, FOP, FCS, FDS, FDP)
+    /// - XMM Registers 0-15
+    /// - MXCSR Control and Status Register
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    fn set_fpu(&self, fpu: &FpuState) -> Result<()>;
+
+    /// Defines the vCPU responses to the CPUID instruction
+    ///
+    /// The information contained in these responses can be set to be consistent
+    /// with hardware, kernel, userspace capabilities, and user requirements.
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    fn set_cpuid2(&self, cpuid: &CpuId) -> Result<()>;
+
+    /// Read the Local APIC state from the vCPU
+    ///
+    /// LAPIC state is comprised of the memory-mapped Local APIC registers.
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    fn get_lapic(&self) -> Result<LapicState>;
+
+    /// Write the Local APIC state to the vCPU
+    ///
+    /// LAPIC state is comprised of the memory-mapped Local APIC registers
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    fn set_lapic(&self, klapic: &LapicState) -> Result<()>;
+
+    /// Read the Model-Specific Registers from the vCPU based on MSR indices
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    fn get_msrs(&self, msrs: &mut MsrEntries) -> Result<(i32)>;
+
+    /// Write the Model-Specific Registers to the vCPU based on MSR indices
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    fn set_msrs(&self, msrs: &MsrEntries) -> Result<()>;
+
+    /// Initialize a vCPU
+    ///
+    /// Specifies the CPU to present to the guest, and the optional features
+    /// it should have
+    #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+    fn vcpu_init(&self, kvi: &kvm_vcpu_init) -> Result<()>;
+
+    /// Set a single register in the virtual CPU
+    #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+    fn set_one_reg(&self, reg_id: u64, data: u64) -> Result<()>;
+
+    /// Run a virtual CPU until there is a VM exit
+    ///
+    /// The output will specify the type of VM exit
+    fn run(&self) -> Result<VcpuExit>;
+
+    /// Get the context from the last vCPU run.
+    ///
+    /// The context contains information related to the latest VM exit
+    fn get_run_context(&self) -> Self::RunContextType;
```

Comment:
The current `Vcpu` trait definition only includes a set of functions implemented by kvm-ioctls. There are, of course, other functions  that would be useful as Vcpu traits (get/set MP state, get/set xsave state, etc.), and any Vcpu not implementing explicitly could use the `unimplemented!()` macro